src/EnergyPlus/ExhaustAirSystemManager.cc

```diff
@@ -0,0 +1,966 @@
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <AirflowNetwork/Elements.hpp>
+#include <EnergyPlus/BranchNodeConnections.hh>
+#include <EnergyPlus/Data/EnergyPlusData.hh>
+#include <EnergyPlus/DataContaminantBalance.hh>
+#include <EnergyPlus/DataEnvironment.hh>
+#include <EnergyPlus/DataHVACGlobals.hh>
+#include <EnergyPlus/DataHeatBalFanSys.hh>
+#include <EnergyPlus/DataHeatBalance.hh>
+#include <EnergyPlus/DataIPShortCuts.hh>
+#include <EnergyPlus/DataLoopNode.hh>
+#include <EnergyPlus/DataSizing.hh>
+#include <EnergyPlus/DataZoneEquipment.hh>
+#include <EnergyPlus/ExhaustAirSystemManager.hh>
+#include <EnergyPlus/Fans.hh>
+#include <EnergyPlus/GeneralRoutines.hh>
+#include <EnergyPlus/HVACFan.hh>
+#include <EnergyPlus/InputProcessing/InputProcessor.hh>
+#include <EnergyPlus/MixerComponent.hh>
+#include <EnergyPlus/NodeInputManager.hh>
+// #include <EnergyPlus/ReturnAirPathManager.hh> //2022-01-14: replace with exhaust system
+#include <EnergyPlus/ScheduleManager.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+// #include <EnergyPlus/ZonePlenum.hh> //2022-01-14: may not needed for this exhaust system
+
+namespace EnergyPlus {
+
+namespace ExhaustAirSystemManager {
+    // Module containing the routines dealing with the AirLoopHVAC:ExhaustSystem
+
+    // struct MixerBranchZone
+    //{
+    //    int mixerInletNodeNum;
+    //    int zoneExhaustNodeNum;
+    //    int zoneNum;
+    //    bool collected;
+
+    //    // default constructor
+    //    // Question: why the constructor skipped the first element std::string Name?
+    //    MixerBranchZone() : mixerInletNodeNum(0), zoneExhaustNodeNum(0), zoneNum(0), collected(false)
+    //    {
+    //    }
+
+    //    MixerBranchZone(int a, int b, int c, bool d) : mixerInletNodeNum(a), zoneExhaustNodeNum(b), zoneNum(c), collected(d)
+    //    {
+    //    }
+    //};
+
+    // std::vector<MixerBranchZone> mixerToZoneTable;
+    // map might be aa better choice:
+    std::map<int, int> mixerBranchMap;
```

Comment:
Actually, `map` is only a better choice when you are dealing with a large number of elements (e.g., >50 or 100). What is this a `map` to and from?

Reply:
This particular line is no longer in use. For the next line `std::map<int, int> mixerIndexMap;` which is still in use, it is mapping from the exhaust control's outlet number (or =the node number of one of the incoming legs of a zone mixer) (i.e. the key) to the exhaust control's index number (i.e. the value) in state.dataZoneEquip->ZoneExhaustControlSystem. So later the exhaust controls' indexes can be directly accessed by using the map keys of zone mixer inlet nodes numbers.

Reply:
I'm not 100% sure what that means, but that's fine for now. This will probably come up again in some future pattern sweep and we can deal with it then. Thank you for being so responsive to all of the other comments.